packages/angular_devkit/build_angular/src/utils/routes-extractor/extractor.ts

```diff
@@ -0,0 +1,102 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import { ApplicationRef, Injector, Type, createPlatformFactory, platformCore } from '@angular/core';
+import {
+  INITIAL_CONFIG,
+  ɵINTERNAL_SERVER_PLATFORM_PROVIDERS as INTERNAL_SERVER_PLATFORM_PROVIDERS,
+} from '@angular/platform-server';
+import { Route, Router, ɵROUTER_PROVIDERS } from '@angular/router';
+import { first } from 'rxjs/operators'; // Import from `/operators` to support rxjs 6 which is still supported by the Framework.
+
+// TODO(alanagius): replace the below once `RouterConfigLoader` is privately exported from `@angular/router`.
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const RouterConfigLoader = ɵROUTER_PROVIDERS[5] as any;
```

Comment:
I'd propose to resolve that asap, since relying on an index would be very unstable. @atscott should we consider exposing a helper method that would take a `Route` as an input and return a list of loaded child `Route`s? That'd probably be the minimal possible API extension to support this. WDYT?

Reply:
I was planning to add a private export to `RouterConfigLoader` when this was approved.

Reply:
We can do that too, I was just wondering if there is a smaller API surface that we can expose (either publicly or via a private symbol), so that it's easier to maintain later in the Router code.

Reply:
Yea, there's certainly a smaller API surface that should be exposed here. There's a _lot_ going on in the config loader including coordinating failed loads, duplicate loads with the preloader, lazy components, dev mode errors, and event emits. There are really only two bits in there you care about: executing the load callback and mapping that result to returned routes (and the injector if it was a lazy module).

Reply:
SGTM, let’s merge as is and get back to this when the API is available.